File: 프로그래머스연습문제/로또의 최고 순위와 최저 순위.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;
    int count = 0;
    int number = 0;
    for(int i = 0; i < lottos.size(); i++){
        for(int j = 0; j < win_nums.size(); j++){
            if(lottos[i] == 0){
                count++;
                break;
            }
            else if(lottos[i] == win_nums[j])number++;
            
        }
    }
    
    if(number == 6){
        answer.push_back(1);
        answer.push_back(1);
    }

    
    else{
        answer.push_back(7-number-count);
        answer.push_back(7-number);
    }

    if (answer[0] == 7)answer[0] = 6;
    if (answer[1] == 7)answer[1] = 6;
    
    
    return answer;
}

File: 프로그래머스_python/Level1/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    answer = []
    
    zero_count = 0
    count = 0
    for i in lottos:
        if i in win_nums:
            win_nums.remove(i)
            count += 1
        elif i == 0:
            zero_count += 1
    
    if count == 0 or count == 1:
        answer.append(min(7 - zero_count - count,6))
        answer.append(min(7 - count,6))
        
    else:
        answer.append(7 - count - zero_count)
        answer.append(7 - count)
    
    
    return answer
